service/service.go

```diff
@@ -388,29 +423,29 @@ func (s *Service) Info() (*InfoResponse, error) {
 }
 
 // newRound creates a new round with the given epoch. This method MUST be guarded by a write lock on openRoundMutex.
-func (s *Service) newRound(epoch uint32) {
+func (s *Service) newRound(ctx context.Context, epoch uint32) {
 	if err := saveState(s.datadir, s.privKey); err != nil {
 		panic(err)
 	}
-	r := newRound(s.sig, s.datadir, epoch)
+	r := newRound(ctx, s.datadir, epoch)
 	if err := r.open(); err != nil {
 		panic(fmt.Errorf("failed to open round: %v", err))
 	}
 
 	s.openRound = r
```

Comment:
assignment without mutex protection or is it not required here?

Reply:
This method assumes to be called under a lock. 
https://github.com/spacemeshos/poet/blob/0a8257879fd821e66791e50b46c3e47edaca5b78/service/service.go#L424-L425